addons/l10n_it_edi/models/account_move.py

```diff
@@ -0,0 +1,1212 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from base64 import b64encode
+from datetime import datetime
+import logging
+from lxml import etree
+from markupsafe import escape
+import uuid
+
+from odoo import _, api, Command, fields, models
+from odoo.addons.base.models.ir_qweb_fields import Markup, nl2br, nl2br_enclose
+from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import AccountEdiProxyError
+from odoo.exceptions import UserError
+from odoo.tools import float_compare, float_repr, cleanup_xml_node
+
+_logger = logging.getLogger(__name__)
+
+
+WAITING_STATES = ['processing', 'forward_attempt']
+
+
+# -------------------------------------------------------------------------
+# XML tool functions
+# -------------------------------------------------------------------------
+
+def get_text(tree, xpath, many=False):
+    texts = [el.text.strip() for el in tree.xpath(xpath) if el.text]
+    return texts if many else texts[0] if texts else ''
+
+def get_float(tree, xpath):
+    return float(get_text(tree, xpath) or '0')
+
+def get_date(tree, xpath):
+    """ Dates in FatturaPA are ISO 8601 date format, pattern '[-]CCYY-MM-DD[Z|(+|-)hh:mm]' """
+    dt = get_datetime(tree, xpath)
+    return dt.date() if dt else False
+
+def get_datetime(tree, xpath):
+    """ Datetimes in FatturaPA are ISO 8601 date format, pattern '[-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]'
+        Python 3.7 -> 3.11 doesn't support 'Z'.
+    """
+    if (datetime_str := get_text(tree, xpath)):
+        try:
+            return datetime.fromisoformat(datetime_str.replace('Z', '+00:00'))
+        except (ValueError, TypeError):
+            return False
+    return False
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_it_edi_state = fields.Selection([
+        ('being_sent', 'Being Sent To SdI'),
+        ('requires_user_signature', 'Requires user signature'),
+        ('processing', 'SdI Processing'),
+        ('rejected', 'SdI Rejected'),
+        ('forwarded', 'SdI Accepted, Forwarded to Partner'),
+        ('forward_failed', 'SdI Accepted, Forward to Partner Failed'),
+        ('forward_attempt', 'SdI Accepted, Forwarding to Partner'),
+        ('accepted_by_pa_partner', 'SdI Accepted, Accepted by the PA Partner'),
+        ('rejected_by_pa_partner', 'SdI Accepted, Rejected by the PA Partner'),
+        ('accepted_by_pa_partner_after_expiry', 'SdI Accepted, PA Partner Expired Terms'),
+    ], copy=False, tracking=True, readonly=True)
```

Comment:
indent the field properly?
```suggestion
    l10n_it_edi_state = fields.Selection(
        selection=[
            ('being_sent', 'Being Sent To SdI'),
            ('requires_user_signature', 'Requires user signature'),
            ('processing', 'SdI Processing'),
            ('rejected', 'SdI Rejected'),
            ('forwarded', 'SdI Accepted, Forwarded to Partner'),
            ('forward_failed', 'SdI Accepted, Forward to Partner Failed'),
            ('forward_attempt', 'SdI Accepted, Forwarding to Partner'),
            ('accepted_by_pa_partner', 'SdI Accepted, Accepted by the PA Partner'),
            ('rejected_by_pa_partner', 'SdI Accepted, Rejected by the PA Partner'),
            ('accepted_by_pa_partner_after_expiry', 'SdI Accepted, PA Partner Expired Terms'),
        ],
        copy=False,
        tracking=True,
        readonly=True,
    )
```